box-ui/src/components/ts/Student/StudentMeeting.tsx

```diff
@@ -1,7 +1,323 @@
+            },
+        }),
+        [],
+    );
+    const state = useLocation().state as LocationState;
+    const [information, setInformation] = useState({
+        roomName: state && state.roomName ? state.roomName : 'dty',
+        domain: state && state.domain ? state.domain : 'meeting.education',
+    });
+
+    const [selectCoord, setSelectCoord] = useState<boolean>(false);
+    const [coord, setCoord] = useState<Array<Array<number>>>([]);
+
+    //circle : svg element to display on click on the image
+    const [circles, setCircles] = useState<React.SVGProps<SVGCircleElement>[]>([]);
+    const [endCarre, setEndCarre] = useState<boolean>(false);
+    const [textBtn, setTextBtn] = useState<string>('Recadrer');
+
+    // img : downloaded from back, potentially cropped
+    const [height_img, setHeight] = useState<number>(0);
+    const [img, setImg] = useState<string>('../../FirstPicture.png');
+
+    // img original : not cropped
+    const [width_img_original, setWidthOriginal] = useState<number>(0);
+    const [height_img_original, setHeightOriginal] = useState<number>(0);
+    const [ratio_img_original, setRatioOriginal] = useState<string>('');
+    const [img_original, setImg_original] = useState<string>('../../FirstPicture.png');
+
+    //Dimension image affichée
+    const [widthAff, setWidthAff] = useState<number>(0);
+    const [heightAff, setHeightAff] = useState<number>(0);
+
+    //Animation loading, waiting for new Image
+    const [loading, setLoading] = useState<boolean>(false);
+
+    useEffect(() => {
+        if (selectCoord) {
+            if (!endCarre) {
+                setTextBtn('Annuler');
+            }
+        } else {
+            setTextBtn('Recadrer');
+            setCoord([]);
+            setEndCarre(false);
+            setCircles([]);
+        }
+    }, [selectCoord]);
+
+    useEffect(() => {
+        getImgSize(img, false);
+    }, [img]);
+
+    useEffect(() => {
+        getImgSize(img_original, true);
+    }, [img_original]);
+
+    useEffect(() => {
+        const ws1 = new WebSocket('ws://localhost:8070/ws1');
+        ws1.onmessage = function (event) {
+            if (event.data == 'true') {
+                // if there is a new image availabe on the back
+                requestImage();
+            }
+        };
+        // ask to backend if there is a new photo to download
+        const interval = setInterval(() => {
+            ws1.send('photo?');
+        }, 1000);
+
+        return () => {
+            ws1.close;
+            clearInterval(interval);
+        };
+    }, []);
+
+    // function called on click to validate the coordinates entry
+    function validerSaisie() {
+        setSelectCoord(false);
+        setEndCarre(false);
+        let c: any[] = [];
+
+        // operation to send the correct coordinates to the back
+        const rapportx = width_img_original / widthAff;
+        const rapporty = height_img_original / heightAff;
+        for (let i = 0; i < 4; i++) {
```

Comment:
```suggestion
        c = coord.map((point) => [point[0] * rapportx, point[1] * rapporty])
```
You could also use a clearer name for `c`